lib/pinboard_tabs_downloader.rb

```diff
@@ -0,0 +1,89 @@
+require 'fileutils'
+require 'pathname'
+require_relative 'downloader/config'
+require_relative 'downloader/scraper'
+
+class PinboardTabsDownloader
+
+  attr_reader :scraper, :all_episodes
+  def initialize
+    @lib_root     = File.expand_path(File.join(File.dirname(__FILE__),'..'))
+    @download_dir = File.join(lib_root, 'pinboard_tabs')
+    @temp_dir     = File.join(lib_root, 'tmp')
+    FileUtils.mkdir_p @download_dir
+    FileUtils.mkdir_p @temp_dir
+    STDOUT.sync = true
+    set_credentials
+    @scraper = RubyTapasDownloader::HtmlScraper.new(@username, @password)
+    log "initialized"
+  end
+
+  def pinboard_url
+    "https://pinboard.in/"
+  end
+  def tabs_url
+    @tabs_url ||= "u:#{@username}/tabs/"
+  end
+
+  def sign_in_url
+    pinboard_url
+  end
+
+  def get_tabs
+    @scraper.sign_in(sign_in_url)
+    tabs_page = @scraper.get(pinboard_url + tabs_url)
+    @tabs_pages ||= tabs_page.search("#main_column td a").map {|a|
+      [a.attribute('href').to_s, a.text]
+    }.reject {|href, text|
+      !href.include?(tabs_url)
+    }
+    self
+  end
+
+  def download
+    @tabs_pages.each do |href, text|
+      puts "Downloading #{href}: #{text}"
+      tab_number = href.split('/')[-1]
+      tab_description = text.gsub(/\s+/, '_').gsub('/', '-')
+      file_name = File.join(@download_dir, "#{tab_number}-#{tab_description}.yaml")
+      next if File.exists?(file_name)
+      links = @scraper.get(href).search("table td a").each_with_object({}) {|a, result|
+        result[a.attribute('href').to_s] = a.text
+      }.reject {|href, text|
```

Comment:
Shadowing outer local variable - `href`.<br>Shadowing outer local variable - `text`.<br>Unused block argument - `text`. If it's necessary, use `_` or `_text` as an argument name to indicate that it won't be used.<br>Avoid using {...} for multi-line blocks.<br>Avoid multi-line chains of blocks.